$OpenBSD: patch-libarchive_archive_write_disk_3,v 1.1 2016/01/30 23:34:44 naddy Exp $

CVE-2015-2304
Commit 5935715
    Add ARCHIVE_EXTRACT_SECURE_NOABSOLUTEPATHS option
    This fixes a directory traversal in the cpio tool.

--- libarchive/archive_write_disk.3.orig	Sun Jan 13 22:10:46 2013
+++ libarchive/archive_write_disk.3	Sat Jan 30 21:33:16 2016
@@ -177,6 +177,9 @@ The default is to not refuse such paths.
 Note that paths ending in
 .Pa ..
 always cause an error, regardless of this flag.
+.It Cm ARCHIVE_EXTRACT_SECURE_NOABSOLUTEPATHS
+Refuse to extract an absolute path.
+The default is to not refuse such paths.
 .It Cm ARCHIVE_EXTRACT_SPARSE
 Scan data for blocks of NUL bytes and try to recreate them with holes.
 This results in sparse files, independent of whether the archive format
